src/binarySearchTree.js

```diff
@@ -0,0 +1,149 @@
+'use strict'
+
+function Node(value){
+  this.value = value
+  this.left = null
+  this.right = null
+  this.parent = null
+  this.leftOrRight = "root"
+}
+
+export default class BinarySearchTree {
+  constructor(){
+    this.root = null
+    this.length = 0
+  }
+
+  search(value){
+    if(!this.root){
+      return null
+    }
+
+    let returnValue = null
+
+    const recursion = (node) => {
+      if(node.value === value){
+        returnValue = node
+        return
+      }
+
+      if(node.left){
+        recursion(node.left)
+        if(returnValue !== null) {
+          return
```

Comment:
Okay if it work's. but if you want we can sit down and find a better way to think about this.